src/main/java/net/neoforged/neoforge/client/textures/FluidSpriteCache.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.textures;
+
+import java.util.Map;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.BlockAndTintGetter;
+import net.minecraft.world.level.material.FluidState;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+import org.jetbrains.annotations.ApiStatus;
+
+public final class FluidSpriteCache {
```

Comment:
Can you add javadoc to this class as well as to `getFluidSprites`? Ideally explaining why this should be used (i.e. off-thread sprite access).

Reply:
I've added some simple docs. I can't think of anything better right now, so if you have suggestions for anything better, then I'm all ears.